packages/gatsby-theme-newrelic/src/components/SwiftSearch.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import SiteSearchAPIConnector from '@elastic/search-ui-site-search-connector';
+import {
+  SearchProvider,
+  WithSearch,
+  SearchBox,
+  Results,
+  Paging,
+  PagingInfo,
+} from '@elastic/react-search-ui';
+import ResultView from './ResultView';
+import PagingInfoView from './PagingInfoView';
+import SearchInput from './SearchInput';
+
+import styled from '@emotion/styled';
+import { css } from '@emotion/core';
+import PropTypes from 'prop-types';
+
+// TODO: move styles to emotion/wrapper
+import Icon from './Icon';
+import styles from '../styles/SwiftSearchStyles';
+
+const connector = new SiteSearchAPIConnector({
+  documentType: 'page',
+  engineKey: 'Ad9HfGjDw4GRkcmJjUut',
+});
+
+const configOptions = {
+  apiConnector: connector,
+  searchQuery: {
+    result_fields: {
+      title: {
+        snippet: {
+          size: 100,
+          fallback: true,
+        },
+      },
+      body: {
+        snippet: {
+          size: 400,
+          fallback: true,
+        },
+      },
+      url: {
+        raw: {},
+      },
+    },
+  },
+  initialState: {
+    resultsPerPage: 10,
+  },
+};
+
+const SwiftSearch = ({ className }) => {
+  return (
+    <div css={styles} className={className}>
+      <SearchProvider config={configOptions}>
+        <WithSearch
+          mapContextToProps={({ isLoading, results, searchTerm }) => ({
+            isLoading,
+            results,
+            searchTerm,
+          })}
+        >
+          {({ isLoading, results, searchTerm }) => {
+            const hasResults = !isLoading && results && results.length > 0;
+            const hasSearched = !isLoading && searchTerm.length > 0;
+            return (
+              <>
+                <SearchBox
+                  searchAsYouType
+                  debounceLength={500}
+                  inputView={InputView}
+                />
+                {isLoading && (
+                  <Icon
+                    css={css`
+                      margin-top: 1rem;
+                    `}
+                    size="1.5rem"
+                    name={Icon.TYPE.LOADER}
```

Comment:
The more I look at this loader, the more I wonder if we should bring in the loader from NR1 (not use the one from the SDK directly, but reimplement it so it looks the same). That being said, I'm not the expert or have the final design say, so my opinion might be dumb 😆 .

Reply:
I did get Daniel's okay on using this one but we can always change it!

Reply:
If Daniel is ok with it, so am I 😄